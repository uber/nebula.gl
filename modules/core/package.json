{
  "name": "nebula.gl",
  "description": "A suite of 3D-enabled data editing overlays, suitable for deck.gl",
  "license": "MIT",
  "version": "0.6.0",
  "author": "Georgios Karnas <georgios@uber.com>",
  "keywords": [
    "webgl",
    "visualization",
    "overlay",
    "layer"
  ],
  "main": "dist/index.js",
  "module": "dist-es6/index.js",
  "files": [
    "dist",
    "dist-es6",
    "src"
  ],
  "scripts": {
    "build-clean": "rm -fr dist dist-es6 && mkdir -p dist dist-es6",
    "build-es6": "rm -fr dist-es6 && BABEL_ENV=es6 babel src --config-file ../../babel.config.js --out-dir dist-es6 --source-maps inline",
    "build-es5": "rm -fr dist && BABEL_ENV=es5 babel src --config-file ../../babel.config.js --out-dir dist --source-maps inline",
    "build": "yarn build-clean && yarn build-es6 && yarn build-es5",
    "flow": "(cd ../.. && yarn flow)",
    "lint": "(cd ../.. && yarn lint)",
    "prepublish": "yarn build",
    "publish-prod": "yarn build && yarn test && yarn test-dist && npm publish",
    "publish-beta": "yarn build && yarn test && yarn test-dist && npm publish --tag beta",
    "test": "yarn lint && yarn build && yarn flow && yarn jest",
    "test-r": "yarn lint && yarn build && yarn test-node && (cd test/rendering-test && node node-rendering.js)",
    "test-fast": "yarn test-node",
    "test-ci": "yarn lint && node test/node-ci.js",
    "test-cover": "NODE_ENV=test tape -r babel-register test/node.js && nyc report",
    "test-node": "(cd ../.. && node test/node.js)",
    "test-dist": "(cd ../.. && node test/node-dist.js)",
    "test-browser": "webpack-dev-server --env.test --progress --hot --open",
    "jest": "(cd ../.. && jest src)",
    "bench": "node test/bench/node.js",
    "bench-browser": "webpack-dev-server --env.bench --progress --hot --open",
    "test-rendering": "(cd test/rendering-test && webpack-dev-server --config webpack.config.test-rendering.js --progress --hot --open)"
  },
  "dependencies": {
    "@deck.gl/experimental-layers": "^6.1.1",
    "@turf/bbox": ">=4.0.0",
    "@turf/bbox-polygon": ">=4.0.0",
    "@turf/bearing": ">=4.0.0",
    "@turf/buffer": ">=4.0.0",
    "@turf/center": ">=4.0.0",
    "@turf/circle": ">=4.0.0",
    "@turf/destination": ">=4.0.0",
    "@turf/difference": ">=4.0.0",
    "@turf/distance": ">=4.0.0",
    "@turf/ellipse": ">=4.0.0",
    "@turf/helpers": ">=4.0.0",
    "@turf/nearest-point-on-line": ">=4.0.0",
    "@turf/point-to-line-distance": ">=4.0.0",
    "@turf/transform-rotate": ">=4.0.0",
    "@turf/transform-translate": ">=4.0.0",
    "cubic-hermite-spline": "^1.0.1",
    "deck.gl": "^6.1.1",
    "geojson-types": "^2.0.1",
    "global": ">=4.3.0",
    "supercluster": ">=2.3.0",
    "viewport-mercator-project": ">=5.0.0"
  },
  "peerDependencies": {}
}
