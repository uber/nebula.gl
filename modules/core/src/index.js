// @flow

export { ArrowStyles, DEFAULT_ARROWS, MAX_ARROWS } from './lib/style';
export { SELECTION_TYPE } from './lib/deck-renderer/deck-drawer';

export { default as Feature } from './lib/feature';
export { default as LayerMouseEvent } from './lib/layer-mouse-event';

export { default as NebulaLayer } from './lib/nebula-layer';
export { default as JunctionsLayer } from './lib/layers/junctions-layer';
export { default as TextsLayer } from './lib/layers/texts-layer';
export { default as SegmentsLayer } from './lib/layers/segments-layer';

export { default as NebulaCore } from './lib/nebula';

// Utils
export { toDeckColor } from './lib/utils';

// Types
export type { Color, Style } from './types';

// Moved to @nebula.gl/layers
export { EditableGeoJsonLayer } from '@nebula.gl/layers';
export { SelectionLayer } from '@nebula.gl/layers';
export { ElevatedEditHandleLayer } from '@nebula.gl/layers';
export { PathOutlineLayer } from '@nebula.gl/layers';
export { PathMarkerLayer } from '@nebula.gl/layers';
export { ModeHandler } from '@nebula.gl/layers';
export { CompositeModeHandler } from '@nebula.gl/layers';
export { ModifyHandler } from '@nebula.gl/layers';
export { DrawPointHandler } from '@nebula.gl/layers';
export { DrawLineStringHandler } from '@nebula.gl/layers';
export { DrawPolygonHandler } from '@nebula.gl/layers';
export { DrawRectangleHandler } from '@nebula.gl/layers';
export { DrawRectangleUsingThreePointsHandler } from '@nebula.gl/layers';
export { DrawCircleFromCenterHandler } from '@nebula.gl/layers';
export { DrawCircleByBoundingBoxHandler } from '@nebula.gl/layers';
export { DrawEllipseByBoundingBoxHandler } from '@nebula.gl/layers';
export { DrawEllipseUsingThreePointsHandler } from '@nebula.gl/layers';
