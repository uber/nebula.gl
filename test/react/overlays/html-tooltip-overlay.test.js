// @flow-ignore
import React from 'react';
import Renderer from 'react-test-renderer';

import { viewport } from '../../core/mocks';
import Nebula from '../../../modules/react/src/lib/nebula-react';
import HtmlTooltipOverlay from '../../../modules/react/src/lib/overlays/html-tooltip-overlay';

it('test HtmlTooltipOverlay no items shown', () => {
  const Component = (
    <Nebula viewport={viewport} layers={[]}>
      <HtmlTooltipOverlay />
    </Nebula>
  );
  const renderer = Renderer.create(Component);
  expect(renderer.toJSON()).toMatchSnapshot();
});
