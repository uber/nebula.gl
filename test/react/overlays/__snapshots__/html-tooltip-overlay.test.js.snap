// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HtmlTooltipOverlay no items shown 1`] = `
<div
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "pointerEvents": "none",
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    id="deckgl-wrapper"
  >
    <canvas
      id="deckgl-overlay"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "height": "100%",
        "pointerEvents": "none",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
</div>
`;
